File: architecture/demo/src/main/java/com/pacific/arch_test_kotlin/di/AppComponent.java
package com.pacific.arch_test_kotlin.di;

import com.pacific.arch_test_kotlin.App;
import javax.inject.Singleton;

import dagger.Component;
import dagger.android.AndroidInjector;
import dagger.android.support.AndroidSupportInjectionModule;

@Singleton
@Component(modules = {
        AppModule.class,
        ActivityBuildersModule.class,
        AndroidSupportInjectionModule.class
})
interface AppComponent extends AndroidInjector<App> {
    @Component.Builder
    abstract class Builder extends AndroidInjector.Builder<App> {
    }
}

File: architecture/example/src/main/java/com/pacific/example/di/AppComponent.kt
package com.pacific.example.di

import com.pacific.arch.example.App
import dagger.Component
import dagger.android.AndroidInjector
import javax.inject.Singleton

@Singleton
@Component(modules = [(AppModule::class)])
interface AppComponent : AndroidInjector<App> {
    @Component.Builder
    abstract class Builder : AndroidInjector.Builder<App>()
}